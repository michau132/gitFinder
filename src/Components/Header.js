import React from 'react';
import PropTypes from 'prop-types';
import {
  TextField,
  Button,
  Tooltip,
  Grid,
} from '@material-ui/core';
import { Search } from '@material-ui/icons';
import styled from 'styled-components';


const StyledTooltip = styled(props => (
  <Tooltip classes={{ popper: props.className, tooltip: 'asd' }} {...props} />
))`
  & .asd {
    font-size: 15px;
    max-width: 600px;
  }
`;

const StyledTextField = styled(TextField)`
  width: calc(100% - 115px);
`;

const StyledHeader = styled.header`
  background-color: #f5f5f5;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2), 0px 4px 5px 0px rgba(0,0,0,0.14), 0px 1px 10px 0px rgba(0,0,0,0.12);
  margin-bottom: 15px;
`;

const Header = ({
  onFormSubmit, inputValue, updateInputValue, errorInput,
}) => (
  <StyledHeader>
    <form
      onSubmit={onFormSubmit}
    >
      <Grid container justify="space-between">
        <StyledTooltip
          open={errorInput}
          title="Invalid github login. In login you can&apos;t have special signs like blank space, @, #, etc."
          disableFocusListener
          disableHoverListener
          disableTouchListener
        >
          <StyledTextField
            type="search"
            error={errorInput}
            onChange={updateInputValue}
            value={inputValue}
            label="Github login"
            variant="outlined"
            color="inherit"
          />
        </StyledTooltip>
        <Button type="submit" variant="contained" color="primary">
            Search
          <Search />
        </Button>
      </Grid>
    </form>
  </StyledHeader>
);

Header.defaultProps = {
  errorInput: null,
};

Header.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
  updateInputValue: PropTypes.func.isRequired,
  inputValue: PropTypes.string.isRequired,
  errorInput: PropTypes.bool,
};

export default Header;
