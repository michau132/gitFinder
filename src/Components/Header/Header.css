.header {
  background-color: #B2AD00;
  color: #7619FF;
  display: grid;
  grid-template-columns: 150px 1fr  ;

}