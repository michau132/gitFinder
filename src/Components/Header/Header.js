import React from 'react';
import FormContainer from '../../Containers/FormContainer';
import styles from './Header.css';

const Header = () => (
    <FormContainer />
);

export default Header;
