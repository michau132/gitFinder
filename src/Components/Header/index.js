import React from 'react';
import PropTypes from 'prop-types';
import styles from './style.css';

const Header = ({ onFormSubmit, historyPush }) => {
  let input;
  return (
    <form
      onSubmit={(e) => {
        e.preventDefault();
        onFormSubmit(input.value);
        historyPush(`/${input.value}`);
      }}
      className={styles.form}
    >
      <input type="text" className={styles.formInput} ref={node => input = node} placeholder="github login" />
      <button type="submit" className={styles.formButton}>Search</button>
    </form>
  );
};

Header.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
  historyPush: PropTypes.func.isRequired,
};

export default Header;
