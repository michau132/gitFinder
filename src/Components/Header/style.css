.form {
    justify-self: center;
    font-weight: bold;
    font-size: 1.1rem;
    height: 10vh;
}

.formTitle {
    margin-left: -80px;
}

.formButton {
    padding: 8px 16px !important;
}


.formInput {
    padding: 8px;
    width: 90%;
}