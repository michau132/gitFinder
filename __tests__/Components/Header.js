import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../../src/Components/Header';


describe('testing Header component', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      onFormSubmit: jest.fn(() => 'onFormSubmit'),
      inputValue: 'adam',
      updateInputValue: jest.fn(() => 'updateInputValue'),
      errorInput: null,
    };
    wrapper = shallow(<Header {...props} />);
  });
  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('submiting form', () => {
    wrapper.simulate('submit');
    expect(props.onFormSubmit).toHaveBeenCalled();
  });
  test('checking form input', () => {
    const input = wrapper.find('input');
    expect(input.prop('value')).toEqual('adam');
    input.simulate('change');
    expect(props.updateInputValue).toHaveBeenCalled();
  });

  test('error is set to null', () => {
    expect(wrapper.find('.error')).toHaveLength(0);
  });
});
