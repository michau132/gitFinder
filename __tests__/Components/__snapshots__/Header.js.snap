// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Header component matching snapshot 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <input
    className=""
    onChange={[MockFunction]}
    placeholder="github login"
    type="text"
    value="adam"
  />
  <Button
    className={null}
    disabled={false}
    onClick={null}
    text="search"
    type="submit"
  />
</form>
`;
